Tools/CISupport/ews-build/is_failing.py

```diff
@@ -0,0 +1,126 @@
+    SUITE = 'layout-tests'
+    THRESHOLD = 10
+
+    @classmethod
+    def summary(cls, test, commit=None, configuration=None):
+        params = dict(**(configuration or dict()))
+        if commit:
+            params['ref'] = commit
+        response = requests.get(
+            '{}/api/results-summary/{}/{}'.format(cls.URL, cls.SUITE, test),
+            params=params,
+        )
+        if response.status_code != 200:
+            return None
+        return response.json()
+
+    @classmethod
+    def expected(cls, test, type=None, commit=None, configuration=None, log=False):
+        data = cls.summary(test, commit=commit, configuration=configuration)
+        if log:
+            print(test)
+            for key, value in (data or dict()).items():
+                if not value:
+                    continue
+                print('    {}: {}%'.format(key, value))
+            if not data:
+                print('    No historic data found for query')
+        if not data:
+            return False
+        if type:
+            return cls.THRESHOLD < data.get(type.lower(), 0)
+        return cls.THRESHOLD < 100 - (data.get('pass', 0) + data.get('warning', 0))
+
+    @classmethod
+    def main(cls, args=None):
+        parser = argparse.ArgumentParser(
+            description='A script which uses the same logic ews-build.webkit.org does to determine if a given ' +
+                        'test in a given configuration on a specific commit is currently expected to fail.',
+        )
+        parser.add_argument(
+            'test', type=str,
+            help='The test to return results for.',
+        )
+        parser.add_argument(
+            '-c', '--commit',
+            type=str, default=None,
+            help='Commit ref to focus on',
+        )
+        parser.add_argument(
+            '-r', '--result',
+            type=str, default=None,
+            help='Result to filter for (failure, crash, timeout, ect.)',
+        )
+        parser.add_argument(
+            '-s', '--style',
+            type=str, default=None,
+            help='Build style configuration to focus on (debug, release, ect.)',
+        )
+        parser.add_argument(
+            '-p', '--platform',
+            type=str, default=None,
+            help='Platform to focus on (mac, ios, gtk, ect.)',
+        )
+        parser.add_argument(
+            '-f', '--flavor',
+            type=str, default=None,
+            help='Flavor to focus on (wk1, wk2, ect.)',
+        )
+        parser.add_argument(
+            '-v', '--version',
+            type=str, default=None,
+            help='OS version focus on (Ventura, Monterey, ect.)',
+        )
+        parser.add_argument(
+            '-a', '--architecture',
+            type=str, default=None,
+            help='Architecture focus on (arm64, x86_64, ect.)',
+        )
+        parsed = parser.parse_args(args)
+
+        configuration = dict()
+        for key in ['style']:
```

Comment:
Nit: might want to add more keys here, although can be done in subsequent PRs as well.